internal/engine/pod.go

```diff
@@ -175,28 +172,15 @@ func maybeTrackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*v
 		runtime.Pods = make(map[k8s.PodID]*v1alpha1.Pod)
 		runtime.PodAncestorUID = matchedAncestorUID
 		ms.RuntimeState = runtime
-
-		// Fall through to the case below to create a new tracked pod.
-	}
-
-	podInfo, ok := runtime.Pods[podID]
-	if !ok {
-		// CASE 2: We have a set of pods for this ancestor UID, but not this
-		// particular pod -- record it
-		podInfo = &v1alpha1.Pod{
-			Name: podID.String(),
-		}
-
-		runtime.Pods[podID] = podInfo
-
-		return podInfo, true
 	}
 
-	// CASE 3: This pod is already in the PodSet, nothing to do.
-	return podInfo, false
+	// Return the existing (if any) and track the new version
```

Comment:
not sure i follow -- why do we return whatever was previously in the map, instead of the current thing we're tracking?

Reply:
The logic before was:
* If already tracked, return ref to that + `isNew: false`
* Otherwise, create stub + `isNew: true`

Then, caller would mutate it a bunch using its event to populate new info, and critically, in the `!isNew` case, look for container restarts by comparing the old and the new.

It was largely this way because the incoming object type (k8s `v1.Pod`) was different than the outgoing object type (`store.Pod`). Now that the incoming + outgoing are both apiserver `v1alpha.Pod`, it makes more sense IMO to just replace the object*, but we still need the old one for that container restart diff logic 😬 

> why do we return whatever was previously in the map, instead of the current thing we're tracking?
So the answer is sorta that previously it was returning both since we just kept mutating it, so would compare containers before overwriting them.

TBH I'm not thrilled with this, but I'm hoping the logic can be further simplified as things move closer to the reconciler model - right now this is both acting as a "reconciler" as well as the downstream consumer, which muddles the waters.


* Ideally we'd treat these as effectively immutable and actually store a `.DeepCopy()` and not let the caller keep mutating it - the only remaining hiccup there is `BaselineRestartCount` which will become the responsibility of the reconciler to populate

Reply:
Gooootcha. then can you mention the reasoning in a comment? E.g.
```Return the existing pod (if any) so we can see if any containers restarted etc., and track the new version```

Reply:
(and maybe mention the motivation in the comment at the top of the func too)